pkg-manager/resolve-dependencies/src/updateProjectManifest.ts

```diff
@@ -168,7 +170,7 @@ function getPrefPreferSpecifiedExoticSpec (
       }
     }
     const selector = versionSelectorType(specWithoutName)
-    if (!selector) {
+    if (!((selector != null) && (selector.type === 'version' || selector.type === 'range')) && opts.isNew === undefined) {
```

Comment:
What about isNew === false? Shouldn't it be just `!opts.isNew`?

Reply:
Since `opts.spacRaw` will both be `ava@latest` in `pnpm update ava` and `pnpm update`,  But their behavior is different:
* `pnpm update ava`:  `latest` updated to  `^5.1.1`
* `pnpm update`: preserve `latest`

I'm not sure what property to use to distinguish them. It happens that `isNew` is undefined in [installCase](https://github.com/pnpm/pnpm/blob/main/pkg-manager/core/src/install/index.ts#L510)(we only set `isNew` in [installSome](https://github.com/pnpm/pnpm/blob/main/pkg-manager/core/src/install/index.ts#L553) method), so I use isNew === undefined to distinguish this scenario.

Reply:
use `opts.isCaseInstall` instead

Reply:
name it `preserveVersionSpec`